acsylla/_cython/statement/statement.pyx

```diff
@@ -60,13 +70,17 @@ cdef class Statement:
     def bind_string(self, str value, int index):
         cdef bytes bytes_value = value.encode()
 
-        self._check_index_or_raise(index)
         self._check_bind_error_or_raise(
             cass_statement_bind_string_n(self.cass_statement, index, bytes_value, len(bytes_value))
         )
 
     def bind_bytes(self, bytes value, int index):
-        self._check_index_or_raise(index)
         self._check_bind_error_or_raise(
             cass_statement_bind_bytes(self.cass_statement, index, <const cass_byte_t*> value, len(value))
         )
+
+
+def create_statement(str statement_str, int parameters=0):
```

Comment:
I wonder if we can easily/quickly calculate the parameters number, and remove this c style call...

Reply:
API is still in a very early stage and not written into stone, so changes can be proposed for changing them, we should only consider the following

- This first release wants to expose a raw API that despite of its verbosity is usable in production.
- The API is kinda an orthogonal projection of the C++ API
- Performance does not get hurt because of the Python wrapper in a substantial way

With that foundations, we would like to have later on another step where a new API would be built on top of the one that we are currently building which would expose a less verbose for using `acsylla` but most likely havins some impact in the performance. For example

```python
class Employee(acsylla.orm.Table):
    id = acsylla.orm.Field.Int(primary_key=true)
    name = acsylla.orm.Field.String()

async def main():
   await Employee.insert(id=1, name="name").with_session(session)
   employee = await Employee.query(id=1).with_session(session).one()
   print(employee.name)
```

This is just an exmple which could be quite far a way on what we get finally, but just wants to show how a potential syntax sugar API might be built on top of the current one, `acsylla.orm.Table` will use behind the scenes the API that we are building right now.

Reply:
my 2 cents, I would think before jumping into ORM, 
just making it a bit more similar to the python-driver options, so it would be familiar enough for who was working with that already.

Reply:
Yeps, making the ORM close to the one already implemented by the Python driver would reduce the pain on migrating from one client to another. Good point.